hazelcast/src/main/java/com/hazelcast/spi/impl/classicscheduler/ScheduleQueue.java

```diff
@@ -0,0 +1,69 @@
+package com.hazelcast.spi.impl.classicscheduler;
+
+/**
+ * The ScheduleQueue is a kind of priority queue where 'tasks' are queued for scheduling.
+ * <p/>
+ * ScheduleQueue support concurrent producers but only need to support single consumers.
```

Comment:
It isn't an implementation concern in the sense that a promise is made that this queue is never accessed by more than one thread. If your implementation can deal with multiple consumers, that is nice. but not needed.